### PR TITLE
Add params check and fix sdk

### PR DESCRIPTION
**Pull request by oo7ww**
1. add params check for contract config
2. add storage history sponsor check in sdk

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-sponsor-faucet/6)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,7 +97,8 @@ The **SponsorFaucet** returns a **rawTx** with suggested gas and input data.
...
-1. ERROR_COLLATERAL_CANNOT_REPLACE_THIRD_PARTY_SPONSOR // 无法替换第三方赞助者
-2. ERROR_COLLATERAL_FAUCET_OUT_OF_MONEY // faucet 余额不足
-3. ERROR_COLLATERAL_SPONSORED_FUND_UNUSED // 已申请赞助未使用
-4. ERROR_COLLATERAL_OVER_COLLATERAL_TOTAL_LIMIT //超过collateral 赞助总额上限
\ No newline at end of file
+1. ERROR_COLLATERAL_CANNOT_REPLACE_OLD_FAUCET // 合约升级，请联系管理员
+2. ERROR_COLLATERAL_CANNOT_REPLACE_THIRD_PARTY_SPONSOR // 无法替换第三方赞助者
+3. ERROR_COLLATERAL_FAUCET_OUT_OF_MONEY // faucet 余额不足
+4. ERROR_COLLATERAL_SPONSORED_FUND_UNUSED // 已申请赞助未使用
+5. ERROR_COLLATERAL_OVER_COLLATERAL_TOTAL_LIMIT //超过collateral 赞助总额上限
```
Comment (boqiu) [inline README.md:104]: Please change the comment of error code to English.

File: faucet.js [hunk 2]
```diff
@@ -16,14 +16,20 @@ class Faucet {
...
+    constructor(url, address, lastAddress) {
         this.cfx = new Conflux({url: url});
         this.address = address;
+        this.lastAddress = lastAddress;
         this.faucet = this.cfx.Contract({
             abi: faucetContract.abi,
             address: address,
         });
+        this.oldFaucet = this.cfx.Contract({
+            abi: faucetContract.abi,
```
Comment (boqiu) [inline faucet.js:30]: Better to add necessary comment that `oldFaucet` and `newFaucet` share the same ABI file, and be careful with compatibility when use `oldFaucet`.

File: faucet.js [hunk 3]
```diff
@@ -66,20 +72,33 @@ class Faucet {
      * @param dapp The address of dapp 
      */
     async checkAppliable(dapp) {
-        let r, sponsorInfo, faucetParams;
+        if(dapp === null) return {flag: false, message:''};
```
Comment (boqiu) [inline faucet.js:75]: Better to add error message, e.g. "contract address not specified"

File: faucet.js [hunk 4]
```diff
@@ -66,20 +72,33 @@ class Faucet {
...
                     flag: false,
                     message: 'ERROR_COLLATERAL_CANNOT_REPLACE_THIRD_PARTY_SPONSOR'
                 }
             }
+
         } catch (e) {
             return {
                 flag: false,
-                message: 'RPC ERROR' + e.toString()
+                message: 'RPC ERROR ' + e.toString()
```
Comment (boqiu) [inline faucet.js:101]: Colon missed, use "RPC ERROR: xxx".

File: faucet.js [hunk 5]
```diff
@@ -66,20 +72,33 @@ class Faucet {
...
+        if(dapp === null) return {flag: false, message:''};
+        let r, sponsorInfo, faucetParams, collateralForStorage, oldDetail;
         try {
-            faucetParams = await this.getFaucetParams();
+            faucetParams = await this.getFaucetParams(dapp);
             sponsorInfo = await this.cfx.getSponsorInfo(dapp);
-            if(sponsorInfo.sponsorForCollateral !== this.address && sponsorInfo.sponsorBalanceForCollateral > faucetParams.collateral_bound) {        
+            collateralForStorage = await this.cfx.getCollateralForStorage(dapp);
+            
+            if(sponsorInfo.sponsorForCollateral === this.lastAddress) {
```
Comment (boqiu) [inline faucet.js:82]: Space missed after `if` keyword, strongly recommend to pay attention to your code style or use tool to format automatically. You're developing a product, not writing a demo.

**Review by boqiu — CHANGES_REQUESTED** [review]

File: faucet.js (hunk 3 above)
Comment (oo7ww) [reply faucet.js]: error message added

File: faucet.js (hunk 4 above)
Comment (oo7ww) [reply faucet.js]: fixed

File: faucet.js (hunk 5 above)
Comment (oo7ww) [reply faucet.js]: Code formated

File: faucet.js (hunk 2 above)
Comment (oo7ww) [reply faucet.js]: ABI is compatible for this use, and a commet is added to explain this.

File: README.md (hunk 1 above)
Comment (oo7ww) [reply README.md]: Modified

**Review by boqiu — APPROVED** [review]

**Pull request merged**